encrypted-media/scripts/requestmediakeysystemaccess.js

```diff
@@ -0,0 +1,311 @@
+function runTest( config, qualifier ) {
+
+    var testname = testnamePrefix( qualifier, config.keysystem ) + ', requestMediaKeySystemAccess';
+
+    function expect_error(keySystem, configurations, expectedError, testname) {
+        promise_test(function(test) {
+            return navigator.requestMediaKeySystemAccess(keySystem, configurations).then(function(a) {
+                assert_unreached('Unexpected requestMediaKeySystemAccess() success.');
+            }, function(e) {
+                assert_equals(e.name, expectedError);
+            });
+        }, testname);
+    }
+
+    function assert_subset(actual, expected, path) {
+        if (typeof expected == 'string') {
+            assert_equals(actual, expected, path);
+        } else {
+            if (expected.hasOwnProperty('length'))
+                assert_equals(actual.length, expected.length, path + '.length');
+            for (property in expected)
+                assert_subset(actual[property], expected[property], path + '.' + property);
+        }
+    }
+
+    function expect_config(keySystem, configurations, expectedConfiguration, testname) {
+        promise_test(function(test) {
+            return navigator.requestMediaKeySystemAccess(keySystem, configurations).then(function(a) {
+                assert_subset(a.getConfiguration(), expectedConfiguration, 'getConfiguration()');
+            });
+        }, testname);
+    }
+
+    // Tests for Key System.
+    expect_error('', [{}], 'InvalidAccessError', 'Empty Key System');
```

Comment:
Could you update the descriptions of the tests? The failures with, for example, 'Empty Key System' or "CODECS=" descriptions would be unclear. It probably would be much better to have all descriptions in format "For %Specific Key System% %Specific error% should be thrown" or "%Specific configuration% should be equal to %Expected configuration%", etc.